### PR TITLE
refactor(headless-client): clean up IPC enums

### PR DESCRIPTION
**Pull request by ReactorScram**
The headless Client doesn't have a corresponding IPC service, so the `TerminatingGracefully` message for it was unused. This PR shuffles the messages around into better categories to get rid of unused cases like that.

This is also a yak shave towards fixing #5453

### CODE REVIEW

File: rust/headless-client/src/lib.rs [hunk 1]
```diff
@@ -89,9 +90,19 @@ pub enum IpcServerMsg {
     TerminatingGracefully,
 }
 
+/// Messages that go all the way from connlib to the GUI process, or to the headless Client
+#[derive(Debug, serde::Deserialize, serde::Serialize)]
+pub enum ConnlibMsgToGui {
```
Comment (ReactorScram) [inline rust/headless-client/src/lib.rs:95]: I'm open to bikeshedding the names here.

I kept `IpcServerMsg` the same. With that `ConnlibMsg` the principle is broadly "messages are named for whence they came" which is maybe not intuitive but it's what I've been using.

File: rust/headless-client/src/main.rs [hunk 2]
```diff
@@ -252,18 +252,15 @@ fn main() -> Result<()> {
 
             match cb {
                 // TODO: Headless Client shouldn't be using messages labelled `Ipc`
-                InternalServerMsg::Ipc(IpcServerMsg::OnDisconnect {
+                ConnlibMsg::ToGui(ConnlibMsgToGui::OnDisconnect {
                     error_msg,
                     is_authentication_error: _,
                 }) => break Err(anyhow!(error_msg).context("Firezone disconnected")),
-                InternalServerMsg::Ipc(IpcServerMsg::OnUpdateResources(_)) => {
```
Comment (ReactorScram) [inline rust/headless-client/src/main.rs:259]: This no longer says "IPC" in a place that isn't doing any IPC 🥳

File: rust/gui-client/src-tauri/src/client/gui.rs [hunk 3]
```diff
@@ -547,6 +548,7 @@ impl Controller {
                 tx.send(self.advanced_settings.clone()).ok();
             }
             Req::Ipc(msg) => self.handle_ipc(msg).await?,
+            Req::IpcClearedLogs => todo!(),
```
Comment (conectado) [inline rust/gui-client/src-tauri/src/client/gui.rs:551]: We should either handle the message on leave it out for a future PR

File: rust/gui-client/src-tauri/src/client/gui.rs [hunk 4]
```diff
@@ -651,10 +653,11 @@ impl Controller {
 
     async fn handle_ipc(&mut self, msg: IpcServerMsg) -> Result<(), Error> {
         match msg {
-            IpcServerMsg::OnDisconnect {
+            IpcServerMsg::ClearedLogs => todo!(),
```
Comment (conectado) [inline rust/gui-client/src-tauri/src/client/gui.rs:656]: Same here

File: rust/headless-client/src/ipc_service.rs [hunk 5]
```diff
@@ -337,6 +336,7 @@ impl<'a> Handler<'a> {
 
     fn handle_ipc_msg(&mut self, msg: ClientMsg) -> Result<()> {
         match msg {
+            ClientMsg::ClearLogs => todo!(),
```
Comment (conectado) [inline rust/headless-client/src/ipc_service.rs:339]: Same here

File: rust/headless-client/src/lib.rs (hunk 1 above)
Comment (conectado) [reply rust/headless-client/src/lib.rs:95]: I propose we inline these into `ConnlibMsg` and `IpcServer`, it involves a bit of duplication but makes thing much easier to read and requires less matching level.s

**Review by conectado — COMMENTED** [review]
Looks good, but I think we need to remove the `todo!`s before merging

**Comment by ReactorScram** [comment]
I'll fold this and the review feedback into #6280 then

**Comment by ReactorScram** [comment]
#6280 supersedes this

**Pull request closed without merging**